openmc/tallies.py

```diff
@@ -1482,19 +1482,30 @@ def _outer_product(self, other, binary_op):
             new_name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
             new_tally.name = new_name
 
+        # Create copies of self and other tallies to rearrange for tally
+        # arithmetic
+        self_copy = copy.deepcopy(self)
+        other_copy = copy.deepcopy(other)
+
         # Find any shared filters between the two tallies
-        self_filters = set(self.filters)
-        other_filters = set(other.filters)
+        self_filters = set(self_copy.filters)
+        other_filters = set(other_copy.filters)
         filter_intersect = self_filters.intersection(other_filters)
 
-        # Align the shared filters to follow in each tally operand
+        # Align the shared filters in successive order
         for i, filter in enumerate(filter_intersect):
-            self_index = self.filters.index(filter)
-            other_filter = other.filters[self_index]
-            if other_filter != filter:
-                other = other.swap_filters(filter, other_filter)
+            self_index = self_copy.filters.index(filter)
+            other_index = other_copy.filters.index(filter)
+
+            # If necessary, swap self filter
+            if self_index != i:
+                self_copy.swap_filters(filter, self_copy.filters[i], inline=True)
```

Comment:
Any reason you chose to use `inline` here as opposed to `inplace`? I was thinking it would nice to mirror the notation used by the Pandas API (*e.g.*, [`pandas.Series.replace`](http://pandas.pydata.org/pandas-docs/version/0.17.0/generated/pandas.Series.replace.html#pandas.Series.replace)) since tally arithmetic essentially turns `Tallies` into glorified (or rather, highly customized) DataFrames.